### PR TITLE
Tizio/fix slopy slopes

### PR DESCRIPTION
**Pull request by TizioMaurizio**
Fixed character sticking to walls with frictionless physics material in AllPlayer->RealityPlayer->RealityBody->CapsuleCollider
Prevented slopes by becoming slippery by doing so
Future bugfix required for slopes >80° that cause the player to be allowed to accelerate infinitely. When he walks towards them he is able to exploit gravity's binding reaction while remaining in the "Air" state

### CODE REVIEW

File: Assets/Code/Scripts/RealityMovement.cs [hunk 1]
```diff
@@ -56,6 +56,16 @@ public class RealityMovement : MonoBehaviour
...
+    //sticking problem with walls was solved by removing collider friction alltogether and reliying only on RealityPlayer's drag, aka how it was probably intended from the beginning
+    //TOREMOVE reality body capsule collider to change its material and fix sticking to walls
+    //private CapsuleCollider _realityBodyCollider;
+
+    //slope gravity vector3
+    private Vector3 _slopeGravity = Vector3.zero;
+    private bool _bodyOnSlope = false;
+    private bool _groundedOnSlope= false;
+
+    public bool Grounded;
```
Comment (stefanofossati) [inline Assets/Code/Scripts/RealityMovement.cs]: Is this variable needed to be public?
The convention is wrong

File: Assets/Code/Scripts/RealityMovement.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public class RealityMovement : MonoBehaviour
 
     [Header("Slope handling")]
     [SerializeField] public float _maxSlopeAngle;
```
Comment (stefanofossati) [inline Assets/Code/Scripts/RealityMovement.cs]: Is this serializedfield variable needed to be public?

File: Assets/Code/Scripts/RealityMovement.cs [hunk 3]
```diff
@@ -221,10 +255,46 @@ private void MovePlayer()
...
+        //tizio: it is actually better to change the friction instead, to avoid side effects
+        //_rigidbody.useGravity = !OnSlope();
+
+        SlopeHandler();
         
     }
 
+    private void SlopeHandler() //works up to 45 degrees because above something strange happens with cosine, ANYWAY player is not supposed to climb such slopes, he's not a steinbock
+    {
+        if (Physics.Raycast(transform.position, Vector3.down, out _slopeHit, _playerHeight * 0.5f + 0.3f))
```
Comment (stefanofossati) [inline Assets/Code/Scripts/RealityMovement.cs]:
```suggestion
        if (Physics.Raycast(_transform.position, Vector3.down, out _slopeHit, _playerHeight * 0.5f + 0.3f))
```

File: Assets/Code/Scripts/RealityMovement.cs [hunk 4]
```diff
@@ -267,17 +337,20 @@ private void ResetJump()
 
     private bool OnSlope()
     {
-        if (Physics.Raycast(transform.position, Vector3.down, out slopeHit, _playerHeight * 0.5f + 0.3f))
+        if (Physics.Raycast(transform.position, Vector3.down, out _slopeHit, _playerHeight * 0.5f + 0.3f))
```
Comment (stefanofossati) [inline Assets/Code/Scripts/RealityMovement.cs]:
```suggestion
        if (Physics.Raycast(_transform.position, Vector3.down, out _slopeHit, _playerHeight * 0.5f + 0.3f))
```

**Comment by stefanofossati** [comment]
The project compiles.
Question: when the rigidbody goes up to the inclined plane and the it stays still, the rigidbody is stationary. In the other case when the rigidbody goes down the inclined plain it goes down without the player interaction. Is this correct?

File: Assets/Code/Scripts/RealityMovement.cs (hunk 2 above)
Comment (TizioMaurizio) [reply Assets/Code/Scripts/RealityMovement.cs]: nope, I left it as it is in develop, I guess it's better to change it to private

**Comment by TizioMaurizio** [comment]
> The project compiles. Question: when the rigidbody goes up to the inclined plane and the it stays still, the rigidbody is stationary. In the other case when the rigidbody goes down the inclined plain it goes down without the player interaction. Is this correct?

I guess it happens because you have too much inertia to be stopped by friction, if you walk back up a bit during that sliding motion the rigidbody should find balance again

**Review by stefanofossati — DISMISSED** [review]
All works fine, before merging, you move the prism in prefabs so thath you shouldn't have conflicts

**Review by stefanofossati — APPROVED** [review]

**Pull request merged**